src/main/java/uk/gov/justice/hmpps/prison/service/ImageService.java

```diff
@@ -39,13 +52,94 @@ public ImageDetail findImageDetail(final Long imageId) {
 
     public Optional<byte[]> getImageContent(final Long imageId, final boolean fullSizeImage) {
         return offenderImageRepository.findById(imageId)
-        .map(i -> fullSizeImage ? i.getFullSizeImage() : i.getThumbnailImage());
+          .map(i -> fullSizeImage ? i.getFullSizeImage() : i.getThumbnailImage());
     }
 
     public Optional<byte[]> getImageContent(final String offenderNo, final boolean fullSizeImage) {
         return offenderImageRepository.findLatestByOffenderNumber(offenderNo)
             .map(i -> fullSizeImage ? i.getFullSizeImage() : i.getThumbnailImage());
     }
 
+    @PreAuthorize("hasRole('IMAGE_UPLOAD')")
+    @VerifyOffenderAccess(overrideRoles = {"IMAGE_UPLOAD"})
+    @HasWriteScope
+    @Transactional
+    public ImageDetail putImageForOffender(final String offenderNumber, final String imageData) {
+        // Uses a 4:3 aspect ratio - will distort square photos! Compact cameras and phones use 4:3 for portrait.
+        final int fullWidth = 427, fullHeight = 570;
+        final int thumbWidth = 150, thumbHeight = 200;
+
+        final var offender = offenderRepository
+            .findOffenderByNomsId(offenderNumber)
+            .orElseThrow(EntityNotFoundException.withMessage(format("No prisoner found for prisoner number %s", offenderNumber)));
+
+        var booking = offender
+            .getLatestBooking()
+            .orElseThrow(EntityNotFoundException.withMessage(format("There are no bookings for %s", offenderNumber)));
+
+        // Set the previously active facial image for this bookingId to inactive
+        var previousImage = offenderImageRepository.findLatestByBookingId(booking.getBookingId());
```

Comment:
Could have used model to set previous to inactive as already loaded OffenderBooking object.  Possible future change